packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart

```diff
@@ -80,64 +108,86 @@ abstract class InAppPurchaseConnection {
 
-  /// Completes a purchase either after delivering the content or the purchase is failed. (iOS only).
+  /// (App Store only) Mark that purchased content has been delivered to the
+  /// user.
   ///
-  /// You are responsible to complete every [PurchaseDetails] whose [PurchaseDetails.status] is [PurchaseStatus.purchased] or [[PurchaseStatus.error].
-  /// Completing a [PurchaseStatus.pending] purchase will cause exception.
+  /// You are responsible for completing every [PurchaseDetails] whose
+  /// [PurchaseDetails.status] is [PurchaseStatus.purchased] or
+  /// [[PurchaseStatus.error]. Completing a [PurchaseStatus.pending] purchase
+  /// will cause an exception.
   ///
-  /// It throws an [UnsupportedError] on Android.
+  /// This throws an [UnsupportedError] on Android.
   Future<void> completePurchase(PurchaseDetails purchase);
 
-  /// Consume a product that is purchased with `purchase` so user can buy it again. (Android only).
+  /// (Play only) Mark that the user has consumed a product.
   ///
-  /// You are responsible to consume purchases for consumable product after delivery the product.
-  /// The user cannot buy the same product again until the purchase of the product is consumed.
+  /// You are responsible for consuming all consumable purchases once they are
+  /// delivered. The user won't be able to buy the same product again until the
+  /// purchase of the product is consumed.
   ///
-  /// It throws an [UnsupportedError] on iOS.
+  /// This throws an [UnsupportedError] on iOS.
   Future<BillingResponse> consumePurchase(PurchaseDetails purchase);
 
-  /// Query all the past purchases.
+  /// Query all previous purchases.
   ///
-  /// The `applicationUserName` is required if you also passed this in when making a purchase.
-  /// If you did not use a `applicationUserName` when creating payments, you can ignore this parameter.
+  /// The `applicationUserName` should match whatever was sent in the initial
+  /// `PurchaseParam`, if anything.
   ///
-  /// For example, when a user installs your APP on a different phone, you want to restore the past purchases and deliver the products that they previously purchased.
-  /// It is mandatory to restore non-consumable and subscription for them; however, for consumable product, it is up to you to decide if you should restore those.
-  /// If you want to restore the consumable product as well, you need to persist consumable product information for your user on your own server and deliver it to them.
+  /// This does not return consumed products. If you want to restore unused
+  /// consumable products, you need to persist consumable product information
+  /// for your user on your own server.
   Future<QueryPurchaseDetailsResponse> queryPastPurchases(
       {String applicationUserName});
 
-  /// A utility method in case there is an issue with getting the verification data originally on iOS.
+  /// (App Store only) retry loading purchase data after an initial failure.
```

Comment:
maybe we should have a `see also` to link this to the [queryPastPurchases]. When users look at the method and see the document, they prob want to know when to actually use it. And we might want to explain they need to use it after failing to get a valid validationData from [queryPastPurchases]. Could be a little wordy but I kinda think it is necessary for user.

Reply:
Done. Also added to `[PurchaseDetails.verificationData]`.